scylla/src/frame/response/cql_to_rust.rs

```diff
@@ -76,6 +84,19 @@ impl<T: FromCQLVal<CQLValue>> FromCQLVal<CQLValue> for Vec<T> {
     }
 }
 
+impl<T1: FromCQLVal<CQLValue> + Eq + Hash, T2: FromCQLVal<CQLValue>> FromCQLVal<CQLValue>
+    for HashMap<T1, T2>
+{
+    fn from_cql(cql_val: CQLValue) -> Result<Self, FromCQLValError> {
+        let mut res = HashMap::with_capacity(64);
+        let vec = cql_val.into_pair_vec().ok_or(FromCQLValError::BadCQLType)?;
```

Comment:
That's not what I meant.

We know how many values we are going to insert into `res`. The idea was to set capacity to the number of elements that we will insert. It's not ideal to set the capacity to a constant value, because:

- If we insert less elements, than we will unnecessarily waste space,
- If we insert more, then we will have to reallocate and rehash the hashmap once or more, which is a little costly.

Although this is a performance optimization which does not affect correctness, we care about performance, and the optimization is quite simple to do.

The two lines above should look like this:

```rust
let vec = cql_val.into_pair_vec().ok_or(FromCQLValError::BadCQLType)?;
let mut res = HashMap::with_capacity(vec.len());
```

Reply:
Done